rclcpp_components/CMakeLists.txt

```
@@ -41,6 +41,11 @@ ament_target_dependencies(component_container
   "rclcpp"
 )
 
+install(FILES 
+  src/node_main.cpp.in
+  DESTINATION ${PROJECT_BINARY_DIR}
```

[`PROJECT_BINARY_DIR`](https://cmake.org/cmake/help/v3.10/variable/PROJECT_BINARY_DIR.html) a the build directory. I think an appropriate install destination is `share`.
```suggestion
  DESTINATION share
```